java-checks/src/main/java/org/sonar/java/checks/security/DisableAutoEscapingCheck.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5247")
+public class DisableAutoEscapingCheck extends IssuableSubscriptionVisitor {
+
+  private static final String MESSAGE = "Make sure disabling auto-escaping feature is safe here.";
+
+  private static final MethodMatchers MUSTACHE_COMPILER_ESCAPE_HTML = MethodMatchers.create()
+    .ofTypes("com.samskivert.mustache.Mustache$Compiler")
+    .names("escapeHTML")
+    .addParametersMatcher("boolean")
+    .build();
+
+  private static final MethodMatchers MUSTACHE_COMPILER_WITH_ESCAPER = MethodMatchers.create()
+    .ofTypes("com.samskivert.mustache.Mustache$Compiler")
+    .names("withEscaper")
+    .addParametersMatcher("com.samskivert.mustache.Mustache$Escaper")
+    .build();
+
+  private static final MethodMatchers MUSTACHE_ESCAPERS_SIMPLE = MethodMatchers.create()
+    .ofTypes("com.samskivert.mustache.Escapers")
+    .names("simple")
+    .withAnyParameters()
+    .build();
+
+  private static final MethodMatchers FREEMARKER_SET_AUTO_ESCAPING_POLICY = MethodMatchers.create()
+    .ofTypes("freemarker.template.Configuration")
+    .names("setAutoEscapingPolicy")
+    .addParametersMatcher("int")
+    .build();
+
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    MethodInvocationTree mit = (MethodInvocationTree) tree;
+    handleJMustache(mit);
+    handleFreeMarker(mit);

```

Comment:
I see the idea, but instead of using a boolean, I moved the `FREEMARKER_SET_AUTO_ESCAPING_POLICY.matches(mit)` from `handleFreeMarker` to here. The final behavior is the same, let me know if it makes sense to you.